api/api.go

```diff
@@ -972,7 +978,9 @@ func (b *BlockChainAPI) getBlockNumber(blockNumberOrHash *rpc.BlockNumberOrHash)
 // it returns nil instead of an error since that is according to the API spec,
 // if the error is not of type ErrNotFound it will return the error and the generic
 // empty type.
-func handleError[T any](err error, log zerolog.Logger) (T, error) {
+func handleError[T any](err error, log zerolog.Logger, collector metrics.Collector) (T, error) {
+	collector.ApiErrorOccurred()
```

Comment:
we should only trigger error metric if there's actually internal error, so only in default case